pkg/identity/restoration/local_identity_restorer.go

```diff
@@ -62,10 +72,31 @@ type LocalIdentityRestorer struct {
 }
 
 func newLocalIdentityRestorer(params localIdentityRestorerParams) *LocalIdentityRestorer {
-	return &LocalIdentityRestorer{
+	restorer := &LocalIdentityRestorer{
 		params:        params,
 		restoredCIDRs: nil, // will be initialized during restoration
 	}
+
+	params.JobGroup.Add(job.OneShot("release-local-identities", func(ctx context.Context, health cell.Health) error {
+		r, err := params.EndpointRestorePromise.Await(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to wait for endpoint restorer promise: %w", err)
+		}
+
+		if err := r.WaitForEndpointRestore(ctx); err != nil {
+			return fmt.Errorf("failed to wait for endpoint restoration: %w", err)
+		}
+
+		// Sleep for the --identity-restore-grace-period (default: 30 seconds k8s, 10 minutes kvstore), allowing
+		// the normal allocation processes to finish, before releasing restored resources.
+		time.Sleep(option.Config.IdentityRestoreGracePeriod)
```

Comment:
I think `option.Config.IdentityRestoreGracePeriod` could be converted to a cell flag now rather than a global one.